cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -782,3 +787,139 @@ profiles:
 		})
 	}
 }
+
+func TestSchedulerFeatureGates(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "scheduler-feature-gates")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	configFeatureGates := filepath.Join(tmpDir, "scheduler_feature_gates.yaml")
+	if err := ioutil.WriteFile(configFeatureGates, []byte(`
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+featureGates:
+  NonPreemptingPriority: true
+  PodOverhead: true
+`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	configUnknownFeatureGates := filepath.Join(tmpDir, "scheduler_unknown_feature_gates.yaml")
+	if err := ioutil.WriteFile(configUnknownFeatureGates, []byte(`
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+featureGates:
+  NonExistingFeature: true
+  PodOverhead: true
+`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []struct {
+		name                 string
+		flags                []string
+		expectedFeatureGates map[string]bool
+		expectedError        string
+	}{
+		{
+			name:                 "by default no feature gates are set",
+			flags:                []string{},
+			expectedFeatureGates: nil,
+		},
+		{
+			name: "feature gates can be set in command-line flags",
+			flags: []string{
+				"--feature-gates", "NonPreemptingPriority=true,PodDisruptionBudget=true",
+			},
+			expectedFeatureGates: map[string]bool{
```

Comment:
nit: s/expectedFeatureGates/wantFeatureGates

Reply:
I'd leave it `expected` as the other tests here use it as well. I prefer consistency. We can change it outside this PR for the whole file later.